phylanx/execution_tree/primitives/base_primitive.hpp

```diff
@@ -177,7 +177,11 @@ namespace phylanx { namespace execution_tree { namespace primitives
         virtual primitive_result_type eval_direct(
             std::vector<primitive_argument_type> const& params) const
         {
-            return eval(params).get();
+            ++count_;
+            hpx::util::high_resolution_timer eval_timer(duration_);
+            const auto& ret = eval(params).get();
```

Comment:
This should read:

    auto && ret = f.get();

otherwise the return value will not be moved.

Also,  using an RAII wrapper to collect the data might be more appropriate (see for instance https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/util/scoped_timer.hpp).

Reply:
Resolved.